app/code/Magento/Catalog/Model/Product/Option.php

```diff
@@ -306,44 +323,30 @@ public function setProduct(Product $product = null)
     /**
      * Get group name of option by given option type
      *
-     * @param string $type
+     * @param string|null $type
      * @return string
      */
-    public function getGroupByType($type = null)
+    public function getGroupByType($type = null): string
     {
         if ($type === null) {
             $type = $this->getType();
         }
-        $optionGroupsToTypes = [
-            self::OPTION_TYPE_FIELD => self::OPTION_GROUP_TEXT,
-            self::OPTION_TYPE_AREA => self::OPTION_GROUP_TEXT,
-            self::OPTION_TYPE_FILE => self::OPTION_GROUP_FILE,
-            self::OPTION_TYPE_DROP_DOWN => self::OPTION_GROUP_SELECT,
-            self::OPTION_TYPE_RADIO => self::OPTION_GROUP_SELECT,
-            self::OPTION_TYPE_CHECKBOX => self::OPTION_GROUP_SELECT,
-            self::OPTION_TYPE_MULTIPLE => self::OPTION_GROUP_SELECT,
-            self::OPTION_TYPE_DATE => self::OPTION_GROUP_DATE,
-            self::OPTION_TYPE_DATE_TIME => self::OPTION_GROUP_DATE,
-            self::OPTION_TYPE_TIME => self::OPTION_GROUP_DATE,
-        ];
 
-        return $optionGroupsToTypes[$type] ?? '';
+        return $this->optionGroupsToTypes[$type] ?? '';
     }
 
     /**
      * Group model factory
      *
      * @param string $type Option type
-     * @return \Magento\Catalog\Model\Product\Option\Type\DefaultType
+     * @return DataObject
```

Comment:
Please revert this change, let’s leave `\Magento\Catalog\Model\Product\Option\Type\DefaultType` there

Reply:
Still not fixed, please check it

Reply:
Sorry, just looked at old code